types/whois-json/index.d.ts

```diff
@@ -0,0 +1,111 @@
+// Type definitions for whois-json 2.0
+// Project: https://github.com/mikemaccana/whois-json
+// Definitions by: Hwai-En Ho <https://github.com/hwaien>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/**
+ * Perform a WHOIS lookup.
+ *
+ * @param domainName - The domain name on which to perform the lookup.
+ * @param options - Options to tweak the behavior of a WHOIS lookup.
+ */
+export default function whois(domainName: string, options?: WhoisOptions): Promise<WhoisResult>;
+
+/** The result of a WHOIS lookup. */
+export interface WhoisResult {
+    domainName: string;
+    registryDomainId: string;
+    registrarWhoisServer: string;
+    registrarUrl: string;
+    updatedDate: string;
+    creationDate: string;
+    registrarRegistrationExpirationDate: string;
+    registrar: string;
+    registrarIanaId: string;
+    registrarAbuseContactEmail: string;
+    registrarAbuseContactPhone: string;
+    domainStatus: string;
+    registrantName: string;
+    registrantOrganization: string;
+    registrantStreet: string;
+    registrantCity: string;
+    registrantStateProvince: string;
+    registrantPostalCode: string;
+    registrantCountry: string;
+    registrantPhone: string;
+    registrantEmail: string;
+    adminName: string;
+    adminOrganization: string;
+    adminStreet: string;
+    adminCity: string;
+    adminStateProvince: string;
+    adminPostalCode: string;
+    adminCountry: string;
+    adminPhone: string;
+    adminEmail: string;
+    techName: string;
+    techOrganization: string;
+    techStreet: string;
+    techCity: string;
+    techStateProvince: string;
+    techPostalCode: string;
+    techCountry: string;
+    techPhone: string;
+    techEmail: string;
+    nameServer: string;
+    dnssec: string;
+    urlOfTheIcannWhoisDataProblemReportingSystem: string;
+    lastUpdateOfWhoisDatabase: string;
+}
+
+/**
+ * The version of SOCKS protocol to exchange network packets between a client and server through a proxy server.
+ */
+export enum SocksProtocolVersion {
+    /**
+     * The SOCKS4 protocol.
+     */
+    SOCKS4 = 4,
+    /**
+     * The SOCKS5 protocol.
+     *
+     * @see - https://tools.ietf.org/html/rfc1928
+     */
+    SOCKS5 = 5
+}
```

Comment:
This declaration doesn't exist at runtime.

Reply:
You are right that it doesn't exist at runtime. I was thinking it is a useful aid at development time to let developers know that only the values 4 and 5 are valid arguments.

Reply:
Yeah - I agree that that would've been nice, but unfortunately this is going to result in users trying to reference a value (`SocksProtocolVersion`) that doesn't actually exist. It also won't allow users to write `4` or `5` directly.

I'd say the unfortunate best that we can ship is something like this:

```ts
type SocksProtocolVersion = 4 | 5;
```

Reply:
Oh didn't know this syntax. I will update it accordingly. On a related note: is there a way to learn TypeScript syntax other than from the handbook? I tried reading the language spec but saw that it is several years old. Any advice is appreciated @DanielRosenwasser :)